src/psycopt2d/feature_transformers.py

```diff
@@ -0,0 +1,83 @@
+"""Contains custom transformers for data preprocessing."""
+from datetime import datetime
+from typing import List, Optional
+
+import numpy as np
+from sklearn.base import BaseEstimator, TransformerMixin
+
+
+class ConvertToBoolean(BaseEstimator, TransformerMixin):
+    def __init__(
+        self,
+        columns_to_skip: Optional[List[str]] = ["age_in_years", "sex_female"],
+        columns_to_include: Optional[List[str]] = None,
+        ignore_dtypes: set = {"datetime64[ns]"},
+    ) -> None:
+        """Convert variables to boolean, used for checking whether a column has
+        a value.
+
+        Args:
+            columns_to_skip (List[str], optional): Columns to not convert to boolean.
+                Defaults to ["age_in_years", "male"].
+            columns_to_include (List[str], optional): Columns to columns_to_include.
+                Default to None in which case all columns are included.
+            ignore_dtypes (set, optional): Data type to not convert. Defaults to
+                {"datetime64[ns]"}.
+        """
+        self.columns_to_skip = columns_to_skip
+        self.columns_to_include = columns_to_include
+        self.ignore_dtypes = ignore_dtypes
+
+    def fit(self, X, y=None):
+        return self
+
+    def transform(self, X, y=None):
+        columns = X.columns
+
+        if self.columns_to_include:
+            columns = [c for c in columns if c in self.columns_to_include]
+        cols_to_round = [
+            c
+            for c in X.columns
+            if (X[c].dtype not in self.ignore_dtype) or c in self.columns_to_skip
+        ]
+
+        for col in cols_to_round:
+            X[col] = X[col].map(lambda x: 1 if x > 0 else np.NaN)
```

Comment:
```suggestion
            X[col] = X[col].map(lambda x: 1 if not pd.isnull(x) else np.NaN)
```

I think this generalises more :-)